json_journal/data_analysis.py

```diff
@@ -82,3 +83,49 @@ def total_profit(full_account_name_balances):
             pass
     total = (income * (-1)) - expenses
     return total
+
+
+def cumulative_profit(ledger):
+    '''
+    Calculate the cumulative profit.
+
+    Parameters
+    ----------
+    Ledger : dict
+        Loaded json file.
+
+    Returns
+    -------
+    Dates : list
+        List of dates.
+    Cumulative_profit : list
+        List of cumulative total of profit.
+    '''
+    sorted_ledger = sorted(ledger, key=lambda d: d['date'])
```

Comment:
I wonder if we should separate this sorting into some pre-conditioning. Perhaps similar to checking that all transactions balance.

Reply:
Agree - I have created a new issue for pre-conditioning which can cover both of these